torch/csrc/jit/tensorexpr/eval.cpp

```diff
@@ -672,13 +672,74 @@ class SimpleIREvaluatorImpl : public IRVisitor {
     return {};
   }
 
+  void check_bounds_throw(int64_t idx, int64_t bound, const BufPtr& buf) {
+    std::stringstream ss;
+    ss << "Index out of bounds in check_bounds. Index: " << idx
+       << "; bounds: [0, " << bound << ").";
+    throw malformed_input(ss.str(), buf);
+  }
+
+  void check_bounds(const BufPtr& buf, const std::vector<ExprPtr>& indices) {
+    const std::vector<ExprPtr>& dims = buf->dims();
+    if (dims.size() != indices.size()) {
+      // indices are flattened, but not buffer
+      if (indices.size() == 1) {
```

Comment:
Nit: From styling perspective it might be better to use assert here instead of if - this would allow us to have 1 less nesting level in the code.